### PR TITLE
fix: copy invite link moves on scroll

### PR DESCRIPTION
**Pull request by gitstart**
# Description

Fixes [#7104]
I fixed the card dropdown menus to ensure their positions are fixed to their parent cards as required

## Demo
[loom video](https://www.loom.com/share/e799d2ad062744a9ab514c364cbb53b5)

## Testing scenarios

- [ ] On the meetings page
  - Click the menu button on any of the meeting cards
  - Scroll up or down
  - The menu should not be fixed to the same screen position, rather, it should move along with its parent card

## Final checklist

- [x] I checked the [code review guidelines](../docs/codeReview.md)
- [ ] I have added [Metrics Representative](../docs/codeReview.md#metrics-representative) as reviewer(s) if my PR invovles metrics/data/analytics related changes
- [x] I have performed a self-review of my code, the same way I'd do it for any other team member
- [ ] I have tested all cases I listed in the testing scenarios and I haven't found any issues or regressions
- [ ] Whenever I took a non-obvious choice I added a comment explaining why I did it this way
- [x] I added the label `One Review Required` if the PR introduces only minor changes, does not contain any architectural changes or does not introduce any new patterns and I think one review is sufficient'
- [x] PR title is human readable and could be used in changelog

### CODE REVIEW

**Comment by gitstart** [comment]
Hi @Dschoordsch, @igorlesnenko, @JimmyLv  this PR is ready for review

File: packages/client/hooks/useMenu.ts
Comment (JimmyLv) [inline packages/client/hooks/useMenu.ts:80]: -1 what's this magic number? 64

File: packages/client/hooks/useMenu.ts
Comment (JimmyLv) [inline packages/client/hooks/useMenu.ts:77]: -1 it's better to extract a type for coords firstly.

File: packages/client/hooks/useMenu.ts
Comment (JimmyLv) [inline packages/client/hooks/useMenu.ts:73]: +1 this could better simplify to `[PortalStatus.Mounted, PortalStatus.Entering, PortalStatus.Entered].includes(portalStatus)`

File: packages/client/hooks/useMenu.ts
Comment (JimmyLv) [reply packages/client/hooks/useMenu.ts:77]: -2 I think you can use `'key' in object`, ref: https://www.freecodecamp.org/news/how-to-check-if-an-object-has-a-key-in-javascript/

File: packages/client/hooks/usePortal.tsx
Comment (JimmyLv) [inline packages/client/hooks/usePortal.tsx:41]: -1 the parent portal is enumerable, if you need to change it to `string` it becomes generic, need to rethink about it

**Review by JimmyLv — CHANGES_REQUESTED** [review]
Thanks! No problem with the functionality, but the code needs to be improved 😁

File: packages/client/hooks/useMenu.ts
Comment (gitstart) [reply packages/client/hooks/useMenu.ts:80]: That was actually a hack we had to use to get the menu in the right position. I will store it in a constant variable for better readability.

File: packages/client/hooks/usePortal.tsx
Comment (gitstart) [reply packages/client/hooks/usePortal.tsx:41]: Never mind, I was able to get a workaround, pushing soon.

File: packages/client/hooks/useMenu.ts
Comment (JimmyLv) [inline packages/client/hooks/useMenu.ts:82]: -1 it's better to refactor as a `guard clauses`, ref: https://refactoring.guru/replace-nested-conditional-with-guard-clauses

**Review by JimmyLv — CHANGES_REQUESTED** [review]

**Review by JimmyLv — APPROVED** [review]

**Comment by mattkrick** [comment]
Thank you for the PR!
Would it be weird in the menu gets scrolled offscreen & the keyboard events still targeted the menu?
Looking around at other menu implementations, I noticed that they try to emulate native behavior by preventing scrolling when a menu is open:

- https://mui.com/material-ui/react-menu/
- https://www.radix-ui.com/docs/primitives/components/dropdown-menu

what do you think about an implementation that did something similar?

**Comment by gitstart-parabol** [comment]
> Thank you for the PR! Would it be weird in the menu gets scrolled offscreen & the keyboard events still targeted the menu? Looking around at other menu implementations, I noticed that they try to emulate native behavior by preventing scrolling when a menu is open:
> 
> * https://mui.com/material-ui/react-menu/
> * https://www.radix-ui.com/docs/primitives/components/dropdown-menu
> 
> what do you think about an implementation that did something similar?

I think it can be done if required, but it's probably not a good UX in my opinion.

**Comment by gitstart-parabol** [comment]
What is the verdict on this? @mattkrick

**Comment by Dschoordsch** [comment]
> Thank you for the PR! Would it be weird in the menu gets scrolled offscreen & the keyboard events still targeted the menu? Looking around at other menu implementations, I noticed that they try to emulate native behavior by preventing scrolling when a menu is open:
> 
> * https://mui.com/material-ui/react-menu/
> * https://www.radix-ui.com/docs/primitives/components/dropdown-menu
> 
> what do you think about an implementation that did something similar?

There are 2 solutions possible and I saw both in the material ui examples:
1) what Matt said: prevent scrolling of the page when the menu is open
2) do not prevent scrolling, but if the focus on the menu changes, scroll it back into view (probably default browser behaviour?)

At the moment keyboard focus seems to be broken on master and on this PR. It would be good to know if this change works well together with #7350.
I did not check yet, but if that would already do 2), then that would be good enough. I feel like 1) is more suited for bigger/more important menus and it would need to give some visual clue why scrolling is not possible anymore so it's not surprising for the user. All this seems overkill in this situation.

**Comment by gitstart-parabol** [comment]
> > Thank you for the PR! Would it be weird in the menu gets scrolled offscreen & the keyboard events still targeted the menu? Looking around at other menu implementations, I noticed that they try to emulate native behavior by preventing scrolling when a menu is open:
> > 
> > * https://mui.com/material-ui/react-menu/
> > * https://www.radix-ui.com/docs/primitives/components/dropdown-menu
> > 
> > what do you think about an implementation that did something similar?
> 
> There are 2 solutions possible and I saw both in the material ui examples:
> 
> 1. what Matt said: prevent scrolling of the page when the menu is open
> 2. do not prevent scrolling, but if the focus on the menu changes, scroll it back into view (probably default browser behaviour?)
> 
> At the moment keyboard focus seems to be broken on master and on this PR. It would be good to know if this change works well together with #7350. I did not check yet, but if that would already do 2), then that would be good enough. I feel like 1) is more suited for bigger/more important menus and it would need to give some visual clue why scrolling is not possible anymore so it's not surprising for the user. All this seems overkill in this situation.

@Dschoordsch I think it would be simpler to merge this PR with #7350 and then, if 2) needs to be implemented, we can raise a separate issue to handle that. What do you think?

**Comment by Dschoordsch** [comment]
Sounds good.

**Comment by Dschoordsch** [comment]
Waiting on #7350 to be merged to ensure these two work well together

**Comment by mattkrick** [comment]
Hi! Thank you for contributing to our app! We're closing this PR due to inactivity, but that doesn't mean the end of the code! This is still a useful feature & if we prioritize it in an upcoming sprint we will use this PR as a reference point.

**Pull request closed without merging**
